src/Plugins/Nop.Plugin.ExternalAuth.Facebook/Controllers/FacebookDataDeletionController.cs

```diff
@@ -0,0 +1,171 @@
+
+                    var signatureRaw = DecodeUrlBase64(split[0]);
+
+                    if (string.IsNullOrEmpty(signatureRaw))
+                        return null;
+
+                    var dataRaw = DecodeUrlBase64(split[1]);
+
+                    if (string.IsNullOrEmpty(dataRaw))
+                        return null;
+
+                    // the decoded signature
+                    var signature = Convert.FromBase64String(signatureRaw);
+                    var dataBuffer = Convert.FromBase64String(dataRaw);
+
+                    // JSON object
+                    var json = Encoding.UTF8.GetString(dataBuffer);
+
+                    var appSecretBytes = Encoding.UTF8.GetBytes(_facebookExternalAuthSettings.ClientSecret);
+                    HMAC hmac = new HMACSHA256(appSecretBytes);
+                    var expectedHash = hmac.ComputeHash(Encoding.UTF8.GetBytes(split[1]));
+
+                    if (!expectedHash.SequenceEqual(signature))
+                    {
+                        await _logger.ErrorAsync($"Facebook datadeletioncallback error: Hashing data disn't match. Data: {json}");
+                        return null;
+                    }
+
+                    var fbUser = JsonConvert.DeserializeObject<FacebookUserDTO>(json);
+                    var authenticationParameters = new ExternalAuthenticationParameters
+                    {
+                        ProviderSystemName = FacebookAuthenticationDefaults.SystemName,
+                        AccessToken = await HttpContext.GetTokenAsync(FacebookDefaults.AuthenticationScheme, "access_token"),
+                        ExternalIdentifier = fbUser.UserId
+                    };
+
+                    var externalAuthenticationRecord = await _externalAuthenticationService.GetExternalAuthenticationRecordByExternalAuthenticationParametersAsync(authenticationParameters);
+
+                    if (externalAuthenticationRecord == null)
+                        return null;
+
+                    var urlHelper = _urlHelperFactory.GetUrlHelper(_actionContextAccessor.ActionContext);
+                    var notificationUrl = urlHelper.RouteUrl(FacebookAuthenticationDefaults.DataDeletionStatusCheckRoute, new { earId = externalAuthenticationRecord.Id }, _webHelper.GetCurrentRequestProtocol());
+
+                    await _logger.InformationAsync($"Facebook datadeletioncallback successful for CustomerId: {externalAuthenticationRecord.CustomerId}, CustomerEmail: {externalAuthenticationRecord.Email}, ExternalAuthenticationRecordId: {externalAuthenticationRecord.Id}, ExternalIdentifier: {externalAuthenticationRecord.ExternalIdentifier}");
+
+                    await _externalAuthenticationService.DeleteExternalAuthenticationRecordAsync(externalAuthenticationRecord);
+
+                    return Ok(new { url = notificationUrl, confirmation_code = $"{fbUser.UserId}" });
+                }
+
+                return null;
```

Comment:
We cannot return just null in the request, it should be some of HTTP status (Ok, I guess, or something their docs require).

Reply:
In their documentation, they have returned null. We have implemented our code as far as their documentation and returning null.